File: 0504-base-7/0504-base-7.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function(num) {
    let ans = ""
    let neg = false
    if(!num){
        return "0"
    }
    if(num<0){
        num*=-1
        neg = true
    }
    while(num != 0){
        ans+= num%7
        num = Math.floor(num/7)
    }
    if(neg){
        ans+="-"
    }
    ans = ans.split("").reverse("").join("")
    return ans
};

File: 0504-base-7/0504-base-7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        ans = "" 
        neg = False
        if(num<0): 
            num*=-1
            neg = True
        while num != 0:
            ans = ans + str(num%7)
            num = num//7
        if(neg):
            ans= ans + "-"
        num = list(ans)
        num.reverse()
        res = "".join(num)
        return res